src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -343,19 +349,12 @@ private static void LogErrors(ScriptHost host)
                 }
             }
 
-            try
-            {
-                // this thread now owns the instance
-                string message = "Stopping Host";
-                instance.TraceWriter?.Info(message);
-                instance.Logger?.LogInformation(message);
+            // this thread now owns the instance
+            string message = "Stopping Host";
+            instance.TraceWriter?.Info(message);
+            instance.Logger?.LogInformation(message);
 
-                await instance.StopAsync();
-            }
-            finally
-            {
-                instance.Dispose();
-            }
+            await StopAndDisposeAsync(instance);
```

Comment:
Locally I'd also see ObjectDisposed exceptions coming from the call to StopAsync. Reusing the StopAndDisposeAsync method here as well.